### ISSUE TITLE
Server thinks player is still in-game when disconnected for hours

### ISSUE THREAD

**Issue by jammet** [cuberite/cuberite#3516]
Client version:   1.8.9
Server OS:       Raspbian
Commit id:      c8b1d8a37d6073786df34836ed1e2c88652ac629

### Expected behavior
Client should have been properly disconnected and the player removed from the server.

### Actual behavior
Haven't been watching, but I am standing next to what appears to be a connected player, also listed in the player list, when this happened, according to the server log below.

Also, for some reason, the "player" was almost buried in spiders attacking it, but probably "invincible". Some of these spiders must have spawned in the well lit, tiny house he was standing in, most were crowding the outside walls.

```
[18:24:27] User Sledgie65 authenticated with UUID 43f2f65b9f2d4c69a8f70b7a247e7e91                                     <W>2017-01-01 19:46:55.779 1 => <4:Black_Lion(15)> Moved Black_Lion:4(15) to Closed Team 3[19:2]
[18:24:27] Player Sledgie65 has joined the game                                                                        <W>2017-01-01 19:49:12.778 1 => <15:Jammet(1)> Moved Jammet:15(1) to Closed Team 2[17:2]
[18:30:12] Kicking player Sledgie65 for "Too many blocks were destroyed per unit time - hacked client?"                <W>2017-01-01 19:49:15.809 1 => <5:Kirauni(29)> Moved Kirauni:5(29) to Closed Team 2[17:2]
[18:30:12] Kicking player Sledgie65 for "Too many blocks were destroyed per unit time - hacked client?"                <W>2017-01-01 19:49:32.901 1 => <5:Kirauni(29)> Kirauni not allowed to Move in Closed Team 3
[18:30:12] Kicking player Sledgie65 for "Too many blocks were destroyed per unit time - hacked client?"                <W>2017-01-01 19:49:56.819 1 => <4:Black_Lion(15)> Moved Black_Lion:4(15) to Closed Team 2[17:2]
[18:30:12] Kicking player Sledgie65 for "Too many blocks were destroyed per unit time - hacked client?"                <W>2017-01-01 22:03:42.275 1 => <5:Kirauni(29)> Connection closed: The remote host closed the connection [1]
[18:30:12] Kicking player Sledgie65 for "Too many blocks were destroyed per unit time - hacked client?"                <W>2017-01-01 22:09:56.228 1 => <3:(-1)> New connection: 84.191.173.195:49733
[18:30:12] Kicking player Sledgie65 for "Too many blocks were destroyed per unit time - hacked client?"                <W>2017-01-01 22:09:56.508 1 => <3:(-1)> Client version 1.3.0 (WinX64: 1.3.0~1820~g4d3d4d8~snapshot)
[18:30:12] Kicking player Sledgie65 for "Too many blocks were destroyed per unit time - hacked client?"                <W>2017-01-01 22:09:56.691 1 => <3:Kirauni(29)> Authenticated
[18:31:16] Clicked: Wrong window ID (exp 31, got 0) received from "Sledgie65"; ignoring click.                         <W>2017-01-01 22:42:11.341 1 => <4:Black_Lion(15)> Connection closed: The remote host closed the connection [1]
[18:53:57] User Sledgie65 authenticated with UUID 43f2f65b9f2d4c69a8f70b7a247e7e91                                     <W>2017-01-01 22:42:13.199 1 => <3:Kirauni(29)> Connection closed: The remote host closed the connection [1]
[18:53:57] Player Sledgie65 has joined the game                                                                        <W>2017-01-01 22:42:14.972 1 => <15:Jammet(1)> Moved Jammet:15(1) to Closed Team 3[19:2]
[19:03:36] Player Sledgie65 has left the game                                                                          <W>2017-01-02 00:07:34.387 1 => <15:Jammet(1)> Connection closed: The remote host closed the connection [1]
[08:27:49] User Jammet authenticated with UUID 5644c5e4e7694fcb8908e56cdcc2d2cd                                        <W>2017-01-02 00:07:35.342 1 => <13:Creysun(39)> Connection closed: The remote host closed the connection [1]
[08:27:49] Player Jammet has joined the game                                                                           <W>2017-01-02 00:07:36.623 1 => <1:Opi(37)> Connection closed: The remote host closed the connection [1]
[08:29:13] [Jammet]: Moin :)                                                                                           <W>2017-01-02 00:07:36.666 1 => Ending voice thread
```

### Steps to reproduce the behavior

Unknown.

**Comment by mathiascode** [comment]
I've experienced this when my computer slows down and Minecraft crashes.

**Comment by LogicParrot** [comment]
https://github.com/cuberite/cuberite/pull/3439 ?

**Comment by mohe2015** [comment]
The answer to everything: #3439

**Comment by madmaxoft** [comment]
Most likely. Let's close this for now.

**Comment by mathiascode** [comment]
It has happened after #3439 was merged. The player that stays online won't disconnect when trying to kick it. It's probably quite hard to find the cause, but at least it doesn't happen often, and the server doesn't crash.

**Comment by madmaxoft** [comment]
Could you get a full memory dump of the server when this is happening? So that we can have a look at the memory state and see what's going on.

**Comment by mathiascode** [comment]
I will provide a memory dump when this happens again.

**Comment by mathiascode** [comment]
Closing in favor of #4483
